devices/webcam/hand_tracking_utility.py

```diff
@@ -0,0 +1,600 @@
+parser.add_argument("-dn", "--device-number", type = int, nargs = 1, help = "specify device number to capture from, default is 0", action = "store", required = False)
+parser.add_argument("-sr", "--sample-rate", type = int, nargs = 1, help = "specify sample rate from device in Hz, default is 30", action = "store", required = False)
+parser.add_argument("-dr", "--device-resolution", type = int, nargs = 2, help = "specify resolution for capture device, default is 854x480", action = "store", required = False)
+parser.add_argument("-ri", "--rescale-image", help = "enable rescaling display window from input size", action = "store_true", required = False)
+parser.add_argument("-rr", "--rescale_resolution", type = int, nargs = 2, help = "specify resolution to rescale display stream to, default is 1280x720", action = "store", required = False)
+
+class PoseFrame(lg.Message):
+    """
+    Represents a single frame of a webcam's stream along with its hand tracking estimations
+
+    @attributes:
+        frame: a (W, H, 3) shape array of the webcam's stream at its default resolution, in the blue/green/red color space
+        landmarks: List[NormalizedLandmarkList], one NormalizedLandmarkList object for each detected hand
+    
+    NormalizedLandmarkList:
+        @attributes:
+            landmark: RepeatedCompositeContainer, list-like container with up to 21 hand pose landmarks with their normalized (X, Y, Z) coordinates as attributes
+    """
+
+    frame: np.ndarray
+    landmarks: List[NormalizedLandmarkList]
+
+class PoseEstimationConfig(lg.Config):
+    """
+    Configuration for MediaPipe's hand pose esitmation graph
+    See https://google.github.io/mediapipe/solutions/hands.html
+
+    @attributes:
+        model_complexity: int
+        min_detection_confidence: float
+        min_tracking_confidence: float
+    """
+
+    model_complexity: int
+    min_detection_confidence: float
+    min_tracking_confidence: float
+
+class PoseEstimationState(lg.State):
+    """
+    State object for hand pose estimation node
+    See https://google.github.io/mediapipe/solutions/hands.html
+
+    @attributes:
+        hands: mediapipe.python.solutions.hands.Hands
+    """
+
+    hands: Optional[HandsType.Hands] = None
+
+class PoseEstimation(lg.Node):
+    """
+    PoseEstimation node, runs VideoFrame streams through MediaPipe's Hands graph to generate pose data
+
+    @topics:
+        INPUT: VideoFrame
+        OUTPUT: PoseFrame
+    
+    @attributes:
+        config: PoseEstimationConfig
+        state: PoseEstimationState
+    """
+
+    INPUT = lg.Topic(VideoFrame)
+    OUTPUT = lg.Topic(PoseFrame)
+    config: PoseEstimationConfig
+    state: PoseEstimationState
+
+    @lg.subscriber(INPUT)
+    @lg.publisher(OUTPUT)
+    async def processFrame(self, message: VideoFrame) -> lg.AsyncPublisher:
+        results: NamedTuple = self.state.hands.process(cv2.cvtColor(message.frame, cv2.COLOR_BGR2RGB))
```

Comment:
Hardcoded cv2.COLOR_BGR2RGB can be made more generic.

Reply:
I'm unsure how to go about making this more generic. CV2 images originate in BGR, while MediaPipe uses RGB format.